src/coreclr/tests/src/CLRTest.Execute.Bash.targets

```diff
@@ -257,7 +257,16 @@ fi
       <PropertyGroup>
       <_CLRTestRunFile Condition="'$(CLRTestIsHosted)'=='true'">"$CORE_ROOT/corerun"</_CLRTestRunFile>
 
-      <BashCLRTestLaunchCmds Condition="'$(CLRTestKind)' == 'BuildAndRun'">
+      <BashCLRTestPreCommands Condition="'$(TargetArchitecture)' == 'wasm'">
+   <![CDATA[
+# Build wasm app containing the test dll
+$(DotnetRoot)/dotnet build /Users/naricc/workspace/runtime-webassembly-ci/src/mono/wasm/wasm-test-runner/WasmTestRunner.csproj /p:TestAssembly=$(OutputPath)/$(MsBuildProjectName).dll /p:TestBinDir=`pwd`
```

Comment:
This path needs fixing, probably by publishing WasmTestRunner.csproj to CORE_ROOT.

Reply:
Do we actually need to build WasmTestRunner.dll in Helix for each test separately? I think that the WasmTestRunner.csproj script conflates two orthogonal aspects - building the "WasmTestRunner.dll" entry point app and using the WasmAppBuilder task to produce the wasm package. I think it would be cleaner to separate these two. We can build WasmTestRunner as part of the Mono product and just publish it to <code>CORE_ROOT</code> with some secondary script "BuildWasmApp.proj" that will call WasmAppBuilder and do the chmod magic.

Reply:
I think doing it that way is possible. Actually, I might be able to get rid of the WasmTestRunner.dll altogeather, as it is not really needed in this design. I will still need the WasmAppBuilder project file, but I think I can just point the WasmAppBuilder MainAssembly parameter at the the test assembly itself.

Reply:
Sounds good to me. By the way, if we own the source code to WasmAppBuilder, could we modify it to include the "chmod" step so that "building a wasm app" becomes basically just a single command-line invocation?

Reply:
I think that is possible, but can we invoke a custom msbuild task like WasmAppBuilder directly on the command line? I'm not sure what that looks like.

We could also invoke it and make the rest of the bash script to the chmod.

Reply:
I see, I didn't realize it was a task. Is there any reason why it must remain an msbuild task? At the first glance I would say that changing it to a standalone app would make it easier to use.

Reply:
A few other things are also using it (at least the libraries tests, and the sample app, maybe others). I'm not sure if they have a hard requirement for it to be a task or not, but at the very least it would require a bunch of changes I don't want to add to this PR.